hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/NestedFieldsTest.java

```diff
@@ -0,0 +1,325 @@
+        final User user = new User(1L, "user1", organization);
+
+        execute("UPDATE test SET name = 'new-name' WHERE __key = 1");
+        assertRowsAnyOrder("SELECT test.id, test.name, test.organization FROM test WHERE __key = 1",
+                rows(3, 1L, "new-name", organization));
+    }
+
+    private User initDefault() {
+        final IMap<Long, User> testMap = instance().getMap("test");
+        execute("CREATE MAPPING test "
+                + "TYPE IMap OPTIONS ("
+                + "'keyFormat'='bigint', "
+                + "'valueFormat'='java', "
+                + "'valueJavaClass'='" + User.class.getName() + "')");
+
+        final Office office = new Office(3L, "office1");
+        final Organization organization = new Organization(2L, "organization1", office);
+        final User user = new User(1L, "user1", organization);
+        testMap.put(1L, user);
+
+        return user;
+    }
+
+    private void execute(String sql, Object ...args) {
+        instance().getSql().execute(sql, args);
+    }
+
+    public static class User implements Serializable {
+        private Long id;
+        private String name;
+        private Organization organization;
+
+        public User() { }
+
+        public User(final Long id, final String name, final Organization organization) {
+            this.id = id;
+            this.name = name;
+            this.organization = organization;
+        }
+
+        public Long getId() {
+            return id;
+        }
+
+        public void setId(final Long id) {
+            this.id = id;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(final String name) {
+            this.name = name;
+        }
+
+        public Organization getOrganization() {
+            return organization;
+        }
+
+        public void setOrganization(final Organization organization) {
+            this.organization = organization;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            final User user = (User) o;
+            return Objects.equals(id, user.id)
+                    && Objects.equals(name, user.name)
+                    && Objects.equals(organization, user.organization);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(id, name, organization);
+        }
+
+        @Override
+        public String toString() {
+            return "User{" +
+                    "id=" + id +
+                    ", name='" + name + '\'' +
+                    ", organization=" + organization +
+                    '}';
+        }
+    }
+
+    public static class Organization implements Serializable, Comparable<Organization> {
```

Comment:
Was surprised to see `Comparable` here. Found out that we need it for the `=` operator, because `ComparisonPredicate` uses `compareTo() == 0` instead of `equals`. I think we should change this. We should use equals, except for `Number` instances, where we should continue using `compareTo()`. It's not common that business classes implement `Comparable`.